### PR TITLE
AEA-816: Add translations for medication list.

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

File: coordinator/src/services/translation/prescription/index.ts [hunk 1]
```diff
@@ -163,33 +166,62 @@ function convertDispensingSitePreference(
...
+function extractMedicationListText(
+  fhirBundle: fhir.Bundle,
+  fhirCommunicationRequests: Array<fhir.CommunicationRequest>
+): Array<core.Text> {
+  return fhirCommunicationRequests
+    .flatMap(communicationRequest => communicationRequest.payload)
+    .filter(isTruthy)
+    .filter(isContentReferencePayload)
+    .map(payload => resolveReference(fhirBundle, payload.contentReference))
+    .flatMap(list => list.entry)
```
Comment (mfjarvis) [inline coordinator/src/services/translation/prescription/index.ts:199]: Should probably check that `entry`, `item` and `display` are all truthy as the validator doesn't enforce this

File: coordinator/tests/services/translation/prescription.spec.ts [hunk 2]
```diff
@@ -107,6 +139,14 @@ describe("PertinentInformation2", () => {
...
   })
 
+  test("Missing contentReference is handled", () => {
+    fhirCommunicationRequests[0].payload.push({contentReference: undefined})
+    const pertinentInformation2Array = convertBundleToPrescription(bundle).pertinentInformation2
+    const firstPertinentInformation2 = pertinentInformation2Array[0]
+    const pertinentInformation1 = firstPertinentInformation2.pertinentLineItem.pertinentInformation1
+    expect(pertinentInformation1).toBeFalsy()
+  })
+
```
Comment (mfjarvis) [inline coordinator/tests/services/translation/prescription.spec.ts:149]: Add tests to check that we handle missing fields in the List resource

File: coordinator/src/services/translation/prescription/index.ts [hunk 3]
```diff
@@ -16,7 +18,7 @@ import {
   getExtensionForUrl,
   getExtensionForUrlOrNull,
   getNumericValueAsString,
-  isTruthy
+  isTruthy, resolveReference
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/services/translation/prescription/index.ts:21]: Cool, thank the IDE for that one

File: coordinator/src/services/translation/prescription/index.ts [hunk 4]
```diff
@@ -163,33 +166,64 @@ function convertDispensingSitePreference(
...
   fhirCommunicationRequests: Array<fhir.CommunicationRequest>,
   fhirMedicationRequests: Array<fhir.MedicationRequest>,
   repeatNumber: core.Interval<core.Timestamp>
 ) {
-  const patientInfoText = extractText(fhirCommunicationRequests)
+  const medicationListText = extractMedicationListText(fhirBundle, fhirCommunicationRequests)
+  const patientInfoText = extractPatientInfoText(fhirCommunicationRequests)
   return fhirMedicationRequests.map((medicationRequest, index) => {
-    const patientInfoTextForLineItem = shouldIncludePatientInfoText(index) ? patientInfoText : []
+    const medicationListTextForLineItem = shouldIncludePatientAdditionalInstructions(index) ? medicationListText : []
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/services/translation/prescription/index.ts:214]: Yeah, not added that behaviour, but can change

[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/services/translation/prescription/index.ts:214]: Not sure if this is better or worse. I think it's a bit worse:
```
function convertPrescriptionPertinentInformation2alt(
  fhirBundle: fhir.Bundle,
  fhirCommunicationRequests: Array<fhir.CommunicationRequest>,
  fhirMedicationRequests: Array<fhir.MedicationRequest>,
  repeatNumber: core.Interval<core.Timestamp>
) {
  const otherMedicationRequests = [...fhirMedicationRequests]
  const firstMedicationRequest = otherMedicationRequests.shift()

  const firstLineItem = convertMedicationRequestToLineItem(
    firstMedicationRequest,
    repeatNumber,
    extractMedicationListText(fhirBundle, fhirCommunicationRequests),
    extractPatientInfoText(fhirCommunicationRequests)
  )

  const otherLineItems = otherMedicationRequests.map(medicationRequest =>
    convertMedicationRequestToLineItem(medicationRequest, repeatNumber, [], [])
  )

  return [
    firstLineItem,
    ...otherLineItems
  ].map(pertinentLineItem => new prescriptions.PrescriptionPertinentInformation2(pertinentLineItem))
}
```

File: coordinator/src/services/translation/prescription/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (dan-jones-iw) [reply coordinator/src/services/translation/prescription/index.ts:21]: is this the point where we `import * as common`

File: coordinator/src/services/translation/prescription/index.ts [hunk 5]
```diff
@@ -5,6 +5,8 @@ import * as prescriptions from "../../../models/hl7-v3/hl7-v3-prescriptions"
 import {DaysSupply, PrescriptionPertinentInformation7, ReviewDate} from "../../../models/hl7-v3/hl7-v3-prescriptions"
 import * as fhir from "../../../models/fhir/fhir-resources"
 import {
+  ContentReferencePayload,
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/prescription/index.ts:8]: just noticed we `import * as fhir` above so can just use `fhir.Content...` in place

File: coordinator/src/services/translation/prescription/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dan-jones-iw) [reply coordinator/src/services/translation/prescription/index.ts:215]: i actually quite like ^ apart from the return statement lol

[the remark this answers is not in the document]
Comment (dan-jones-iw) [reply coordinator/src/services/translation/prescription/index.ts:215]: only based on looks tho i think it reads quite well

[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/services/translation/prescription/index.ts:215]: I think it's nicer if the first item check isn't in its own method:
```
function convertPrescriptionPertinentInformation2(
  fhirBundle: fhir.Bundle,
  fhirCommunicationRequests: Array<fhir.CommunicationRequest>,
  fhirMedicationRequests: Array<fhir.MedicationRequest>,
  repeatNumber: core.Interval<core.Timestamp>
) {
  return fhirMedicationRequests.map((medicationRequest, index) => {
    const isFirstItem = index === 0
    const pertinentLineItem = convertMedicationRequestToLineItem(
      medicationRequest,
      repeatNumber,
      isFirstItem ? extractMedicationListText(fhirBundle, fhirCommunicationRequests) : [],
      isFirstItem ? extractPatientInfoText(fhirCommunicationRequests) : []
    )
    return new prescriptions.PrescriptionPertinentInformation2(pertinentLineItem)
  })
}
```

[the remark this answers is not in the document]
Comment (mfjarvis) [reply coordinator/src/services/translation/prescription/index.ts:215]: But more than happy to process the first one and the rest separately if you think that's clearer :+1:

File: coordinator/src/services/translation/prescription/index.ts (hunk 5 above)
Comment (mfjarvis) [reply coordinator/src/services/translation/prescription/index.ts]: Done

File: coordinator/src/services/translation/prescription/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dan-jones-iw) [reply coordinator/src/services/translation/prescription/index.ts:207]: feel like the first one makes more explicit/obvious that the first line item is a special case but either look good imo

**Review by booshibooshi — APPROVED** [review]

**Pull request merged**